src/cls/rgw/cls_rgw.cc

```diff
@@ -862,6 +956,10 @@ int rgw_bucket_prepare_op(cls_method_context_t hctx, bufferlist *in, bufferlist
 		 __func__, escape_str(idx).c_str(), rc);
     return rc;
   }
+  if (header.resharding_in_logrecord()) {
+    // write the duplicated index entry copy
+    return reshard_log_index_operation(hctx, idx, entry.key, &info_bl);
+  }
```

Comment:
can replace this and the `cls_cxx_map_set_val()` call above with `write_entry()`